istioctl/cmd/proxyconfig.go

```diff
@@ -365,7 +377,18 @@ func listenerConfigCmd() *cobra.Command {
 			var configWriter *configdump.ConfigWriter
 			var err error
 			if len(args) == 1 {
-				podName, ns := handlers.InferPodInfo(args[0], handlers.HandleNamespace(namespace, defaultNamespace))
+				var client kube.ExtendedClient
+				client, err = kubeClient(kubeconfig, configContext)
+				if err != nil {
+					return fmt.Errorf("failed to create k8s client: %w", err)
+				}
+				var podName, ns string
```

Comment:
nit: Maybe move all the declaration in one place?

```go
var (
	configWriter *configdump.ConfigWriter
	err error
	client kube.ExtendedClient
	podName, ns string
)
```

Reply:
As part of the refactor I moved podName and ns (renamed PodNamespace) to their own section, eliminated err and client, and left configWriter alone.